From f8cc7c17c93afd18bab9b501f35d845da2d098d9 Mon Sep 17 00:00:00 2001
From: Stephen Reynolds <Stephen.Reynolds@abaco.com>
Date: Mon, 3 Dec 2018 11:34:37 +0000
Subject: [PATCH] fall through to use 11A case

---
 setup.bash | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/setup.bash b/setup.bash
index 536b3bc..05db246 100755
--- a/setup.bash
+++ b/setup.bash
@@ -53,9 +53,8 @@ portfromboard() {
 	case $board in 
 
     SBC314)
-		port="/dev/ttyS4"
-		;;
-
+		#drop through to same as 11A
+		;&
     PPC11A)
 		# on the 11a we have to go looking for the 
 		#correct terminal port to use
-- 
1.8.3.1

